### PR TITLE
Explore: props automatically inferred from mapStateToProps and mapDispatchToProps

### LINKED ISSUE #35099: Explore: Refactor typings for redux-connected components
We should refactor the way the props for components are defined so that they can be automatically inferred from `mapStateToProps` s functions.

Example: https://github.com/grafana/grafana/blob/a55a2722764fdae5b7091b432bfc6e0cd1d61f1b/public/app/features/explore/ExplorePaneContainer.tsx#L125

See: https://github.com/grafana/grafana/blob/main/contribute/style-guides/redux.md#typing-of-connected-props

Here's a list of components that need refactoring. Check them off when they've been refactored

**Components**
inside `public/app/features/explore`
- [x] NodeGraphContainer.tsx
- [x] ReturnToDashboardButton.tsx
- [x] TableContainer.tsx

### PR DESCRIPTION
**Pull request by olbo98**
**What this PR does / why we need it**:
This PR adds refactoring so that props are now automatically inferred from matchStateToProps and mapDispatchToProps. The props have been split into explicit props `ExploreProps` and implicit props that are mapped to the component from the store.
See: https://github.com/grafana/grafana/blob/main/contribute/style-guides/redux.md#typing-of-connected-props

**Which issue(s) this PR fixes**:

Fixes #35099

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by olbo98** [comment]
@Elfo404 on line 417-432 in `mapStateToProps` we return a lot of props which are also present in `ExploreProps`. Should these be removed from `ExploreProps` since they are already inferred via `mapStateToProps`?

**Comment by zoltanbedi** [comment]
This is a good start! I love these refactorings! You found out perfectly that we don't need those props.  One thing I'd not name the Props to Props in a case when it is exported. It could cause trouble in other components. I would use the Explore props and extend it with the connector like in https://github.com/grafana/grafana/blob/olof/explore-refactor-typings-redux/public/app/features/explore/LogsContainer.tsx#L189
As far as I saw we will only need exploreId and theme to be in the ExploreProps.

**Comment by olbo98** [comment]
@zoltanbedi Awesome! Thank you for the info, i'll look into it and add those changes before merging

**Comment by olbo98** [comment]
@grafana/observability-squad Could someone help me figure out why the build is failing?

**Comment by ifrost** [comment]
> @zoltanbedi Awesome! Thank you for the info, i'll look into it and add those changes before merging

Weird 🤔 doesn't seem to be related to your changes. I've restarted the build.

File: public/app/features/explore/Explore.tsx [hunk 1]
```diff
@@ -71,36 +60,12 @@ const getStyles = stylesFactory((theme: GrafanaTheme) => {
...
-  isLive: boolean;
-  syncedTimes: boolean;
-  updateTimeRange: typeof updateTimeRange;
-  graphResult: DataFrame[] | null;
-  logsResult?: LogsModel;
-  absoluteRange: AbsoluteTimeRange;
-  timeZone: TimeZone;
   onHiddenSeriesChanged?: (hiddenSeries: string[]) => void;
-  queryResponse: PanelData;
   originPanelId: number;
```
Comment (zoltanbedi) [inline public/app/features/explore/Explore.tsx:67]: I think `originPanelId, onHiddenSeriesChanged, scanRange, scanning` are still can be removed.

**Review by zoltanbedi — CHANGES_REQUESTED** [review]

File: public/app/features/explore/Explore.tsx (hunk 1 above)
Comment (olbo98) [reply public/app/features/explore/Explore.tsx]: Yeah you're right, removed them now 👍

**Review by zoltanbedi — APPROVED** [review]
Good job! 🎉

**Pull request merged**